File: Quorum3Plugins/GamePlugin/JavaGame/src/plugins/quorum/Libraries/Game/Graphics/TextureAttribute.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.Game.Graphics;

/**
 *
 * @author alleew
 */
public class TextureAttribute extends Attribute
{
    public java.lang.Object me_ = null;
    
    public static final int DIFFUSE_TEXTURE = RegisterStatic("diffuseTexture");
    public static final int SPECULAR_TEXTURE = RegisterStatic("specularTexture");
    public static final int BUMP_TEXTURE = RegisterStatic("bumpTexture");
    public static final int NORMAL_TEXTURE = RegisterStatic("normalTexture");
    public static final int AMBIENT_TEXTURE = RegisterStatic("ambientTexture");
    public static final int EMISSIVE_TEXTURE = RegisterStatic("emissiveTexture");
    public static final int REFLECTION_TEXTURE = RegisterStatic("reflectionTexture");
    public static final int MASK = DIFFUSE_TEXTURE | SPECULAR_TEXTURE | BUMP_TEXTURE | NORMAL_TEXTURE | AMBIENT_TEXTURE | EMISSIVE_TEXTURE | REFLECTION_TEXTURE;
    
    public int GetDiffuseValue()
    {
        return DIFFUSE_TEXTURE;
    }
    
    public int GetSpecularValue()
    {
        return SPECULAR_TEXTURE;
    }
    
    public int GetBumpValue()
    {
        return BUMP_TEXTURE;
    }
    
    public int GetNormalValue()
    {
        return NORMAL_TEXTURE;
    }
    
    public int GetAmbientValue()
    {
        return AMBIENT_TEXTURE;
    }
    
    public int GetEmissiveValue()
    {
        return EMISSIVE_TEXTURE;
    }
    
    public int GetReflectionValue()
    {
        return REFLECTION_TEXTURE;
    }
    
    public int GetAttributeMask()
    {
        return MASK;
    }
    
    public boolean SupportsAttribute(int type)
    {
        return (type & MASK) != 0;
    }
}

File: Quorum/Library/Standard/Plugins/javascript/Libraries/Game/Graphics/TextureAttribute.js
function plugins_quorum_Libraries_Game_Graphics_TextureAttribute_(quorumObject) 
{
    var attributePlugin = quorumObject.Libraries_Game_Graphics_Attribute__.plugin_;
    
    this.DIFFUSE_TEXTURE = attributePlugin.Register("diffuseTexture");
    this.SPECULAR_TEXTURE = attributePlugin.Register("specularTexture");
    this.BUMP_TEXTURE = attributePlugin.Register("bumpTexture");
    this.NORMAL_TEXTURE = attributePlugin.Register("normalTexture");
    this.AMBIENT_TEXTURE = attributePlugin.Register("ambientTexture");
    this.EMISSIVE_TEXTURE = attributePlugin.Register("emissiveTexture");
    this.REFLECTION_TEXTURE = attributePlugin.Register("reflectionTexture");
    this.MASK = this.DIFFUSE_TEXTURE | this.SPECULAR_TEXTURE | this.BUMP_TEXTURE
        | this.NORMAL_TEXTURE | this.AMBIENT_TEXTURE | this.EMISSIVE_TEXTURE | this.REFLECTION_TEXTURE;
    
    this.SupportsAttribute$quorum_integer = function(type) 
    {
        return (type & this.MASK) !== 0;
    };
    
    this.GetDiffuseValue = function() 
    {
        return this.DIFFUSE_TEXTURE;
    };
    
    this.GetSpecularValue = function() 
    {
        return this.SPECULAR_TEXTURE;
    };
    
    this.GetBumpValue = function() 
    {
        return this.BUMP_TEXTURE;
    };
    
    this.GetNormalValue = function() 
    {
        return this.NORMAL_TEXTURE;
    };
    
    this.GetAmbientValue = function() 
    {
        return this.AMBIENT_TEXTURE;
    };
    
    this.GetEmissiveValue = function() 
    {
        return this.EMISSIVE_TEXTURE;
    };
    
    this.GetReflectionValue = function() 
    {
        return this.REFLECTION_TEXTURE;
    };
    
    this.GetAttributeMask = function() 
    {
        return this.MASK;
    };
    
    this.GetAttributeType$quorum_text = function(alias)
    {
        attributePlugin.GetAttributeType$quorum_text(alias);
    };
    
    this.GetAttributeAlias$quorum_integer = function(type)
    {
        attributePlugin.GetAttributeAlias$quorum_integer(type);
    };
    
    this.Register = function(alias)
    {
        attributePlugin.Register(alias);
    };
}
